scipy/special/_special_ufuncs_docs.cpp

```diff
@@ -4927,6 +4963,42 @@ const char *mathieu_sem_doc = R"(
     --------
     mathieu_a, mathieu_b, mathieu_cem
 
+    Notes
+    -----
+    Odd Mathieu functions are the solutions to Mathieu's differential equation
+
+    .. math::
+
+        \frac{d^2y}{dx^2} + (a_m - 2q \cos(2x))y = 0
+
+    for which the characteristic number :math:`a_m` results in an odd, periodic
```

Comment:
```suggestion
    for which the characteristic number :math:`b_m` results in an odd, periodic
```
I thought about explicitly mentioning that characteristic values can be found with `mathieu_b` here (and `mathieu_a` in `mathieu_cem`), but I'll wait for feedback before doing that.

Reply:
As a concrete suggestion
```suggestion
    for which the characteristic number :math:`b_m`
    (which can be calculated with `mathieu_b`)
    results in an odd, periodic
```
and the change suggested [above](https://github.com/scipy/scipy/pull/21578/files#r1764484030).
(I'd make an equivalent change in the `mathieu_cem` documentation about finding the characteristic number $a_m$ with `mathieu_a`.)

Do you think I should add these changes, @stefanv? Happy to leave it as is if you don't think these changes are helpful overall. The intent is just to de-mystify these magic numbers $a_m$ that we don't provide as an argument to `mathieu_cem`/`mathieu_sem` directly and to establish the connection among some of the `mathieu`-prefixed functions.

Reply:
Sure, why not, rather be explicit. I'd maybe just put "calculated with `mathieu_b`" in brackets to keep the sentence concise.

Reply:
Good idea. Done!

Reply:
Looks very nice, thanks!